autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/ui/credential/management/viewing/AutofillManagementCredentialsMode.kt

```diff
@@ -125,13 +137,27 @@ class AutofillManagementCredentialsMode : DuckDuckGoFragment(R.layout.fragment_a
         super.onViewCreated(view, savedInstanceState)
         requireActivity().addMenuProvider(this)
         observeViewModel()
-        binding.watchSaveState(saveStateWatcher) {
-            viewModel.allowSaveInEditMode(it)
-        }
         configureUiEventHandlers()
         initialiseToolbar()
     }
 
+    private fun startEditTextWatchers() {
+        val initialState = binding.currentTextState()
+        viewModel.allowSaveInEditMode(false)
+        binding.watchSaveState(saveStateWatcher) {
+            val currentState = binding.currentTextState()
+
+            val changed = currentState != initialState
+            val empty = currentState.isEmpty()
+
+            viewModel.allowSaveInEditMode(!empty && changed)
```

Comment:
Is it necessary to check if the data has changed? From what I see, we only hide the check icon when all fields are empty otherwise we show it. Even on edit mode, we show it right away even if there were no changes.

Reply:
Ah okay, now I noticed that the changed is only correctly checked when:
- List mode -> click on item -> credential mode -> edit.

However, the check is not taking effect for:
- List mode -> context menu on item -> edit